ams/server/src/main/java/com/netease/arctic/server/manager/SpiPluginManager.java

```diff
@@ -66,12 +69,27 @@ public void install(String pluginName) {
 
   @Override
   public void uninstall(String pluginName) {
+    PreconditionUtils.checkExist(installedPlugins.containsKey(pluginName),
+        "Plugin " + pluginName);
     installedPlugins.remove(pluginName);
   }
 
+  @NotNull
   @Override
-  public List<T> list() {
-    return new ArrayList<>(installedPlugins.values());
+  public Iterator<T> iterator() {
+    return new Iterator<T>() {
```

Comment:
Can return 'return installedPlugins.values().iterator()' dirctly

Reply:
the same as above, unmodifiable

Reply:
get